src/Engine/Data/DrawPrimitives.cpp

```diff
@@ -253,9 +253,21 @@ MeshPtr Sphere( const Core::Vector3& center, Scalar radius, const Core::Utils::C
     std::for_each( data.begin(), data.end(), [center]( Core::Vector3& v ) { v += center; } );
     vertices.unlock();
 
-    geom.addAttrib(
-        Ra::Core::Geometry::getAttribName( Ra::Core::Geometry::MeshAttrib::VERTEX_COLOR ),
-        Core::Vector4Array { geom.vertices().size(), color } );
+    return make_shared<Mesh>( "Sphere Primitive", std::move( geom ) );
+}
+
+MeshPtr ParametricSphere( const Core::Vector3& center,
+                          Scalar radius,
+                          const Core::Utils::Color& color,
+                          bool generateTexCoord ) {
+    auto geom   = makeParametricSphere<32, 32>( radius, color, generateTexCoord );
```

Comment:
I think we should avoid hard-coded litteral constants and rather uses typed constants (i.e `constexpr const int slices = 32;`)

Reply:
Same as above, I think we will remove these call in a near futur. But improvement welcome if you want to fix these hard codded values.